### PR TITLE
Fix build failure with SimpleKeychain 0.11.0.

### PR DESCRIPTION
**Pull request by jshier**
### Changes

Fix call to `keyRefOfRSAKey` to properly handle the optional return from SimpleKeychain 0.11.0.

### References

For package managers pulling an unconstrained version of SimpleKeychain (e.g. CocoaPods), Auth0's build is currently broken due to a mismatch in expected optionality between the two libraries.

### Testing

Integrating this branch through CocoaPods should catch the build issue.

### Checklist

* [x] I have read the [Auth0 general contribution guidelines](https://github.com/auth0/open-source-template/blob/master/GENERAL-CONTRIBUTING.md)
* [x] All existing and new tests complete without errors
* [ ] All active GitHub checks have passed

### CODE REVIEW

File: Cartfile [hunk 1]
```diff
@@ -1,2 +1,2 @@
-github "auth0/SimpleKeychain" 
-github "auth0/JWTDecode.swift"
\ No newline at end of file
+github "auth0/SimpleKeychain" >= 0.11.0
```
Comment (Widcket) [inline Cartfile]: Please leave it unconstrained.

File: Auth0.podspec [hunk 2]
```diff
@@ -62,18 +62,18 @@ Pod::Spec.new do |s|
...
-  s.osx.dependency 'SimpleKeychain'
+  s.osx.dependency 'SimpleKeychain', '~> 0.11.0'
   s.osx.dependency 'JWTDecode'
   s.watchos.source_files = 'Auth0/*.swift'
   s.watchos.exclude_files = watchos_exclude_files
   s.watchos.dependency 'JWTDecode'
   s.tvos.source_files = 'Auth0/*.swift'
   s.tvos.exclude_files = web_auth_files
-  s.tvos.dependency 'SimpleKeychain'
+  s.tvos.dependency 'SimpleKeychain', '~> 0.11.0'
```
Comment (Widcket) [inline Auth0.podspec]: Please leave it unconstrained (also all the instances above).

**Comment by jshier** [comment]
Reverted changes to minimum versions for package managers.

File: Cartfile (hunk 1 above)
Comment (jshier) [reply Cartfile]: Reverted.

File: Auth0.podspec (hunk 2 above)
Comment (jshier) [reply Auth0.podspec]: Reverted.

**Comment by Widcket** [comment]
Released, thanks for the PR!

**Comment by jshier** [comment]
Great, thanks!

**Pull request merged**